sklearn/decomposition/nmf.py

```diff
@@ -986,6 +986,11 @@ def non_negative_factorization(X, W=None, H=None, n_components=None,
     factorization with the beta-divergence. Neural Computation, 23(9).
     """
 
+    if init == "warn":
+        warnings.warn("The default value of init will change from "
+                      "random to None in 0.22.", FutureWarning)
```

Comment:
0.23

Reply:
The behaviour won't change if `n_components >= n_features` so we could add that condition